WpfAntSimulator/MainWindow.xaml.cs

```diff
@@ -186,5 +189,49 @@ private BitmapImage Bitmap2BitmapImage(Bitmap bitmap)
                 return bitmapImage;
             }
         }
+        private static bool IsTextAllowed(string text)
+        {
+            return !_regex.IsMatch(text);
+        }
+
+        private void PreviewTextInput(object sender, TextCompositionEventArgs e)
+        {
+            e.Handled = !IsTextAllowed(e.Text);
+        }
+
+        private void SelectObstacle_Click(object sender, RoutedEventArgs e)
+        {
+            selectedObject = new Obstacle();
```

Comment:
Why are you creating a new obstacle object if you mean to select one that already exists.